contracts/extensions/StreamingFeeSplitExtension.sol

```diff
@@ -0,0 +1,206 @@
+
+import { IStreamingFeeModule } from "../interfaces/IStreamingFeeModuleV2.sol";
+
+/**
+ * @title StreamingFeeSplitExtension
+ * @author Set Protocol
+ *
+ * Smart contract global extension which provides DelegatedManager owner and 
+ * methodologist the ability to accrue and split streaming fees at an mutable percentage.
+ */
+contract StreamingFeeSplitExtension is BaseGlobalExtension {
+    using Address for address;
+    using PreciseUnitMath for uint256;
+    using SafeMath for uint256;
+
+    /* ============ Events ============ */
+
+    event ExtensionInitialized(
+        address _setToken,
+        address _delegatedManager
+    );
+
+    event FeesDistributed(
+        address _setToken,
+        address indexed _ownerFeeRecipient,
+        address indexed _methodologist,
+        uint256 _ownerTake,
+        uint256 _methodologistTake
+    );
+
+    /* ============ State Variables ============ */
+
+    // Instance of StreamingFeeModule
+    IStreamingFeeModule public immutable streamingFeeModule;
+
+    // Mapping from Set Token to DelegatedManager 
+    mapping(ISetToken => IDelegatedManager) public setManagers;
+
+    /* ============ Constructor ============ */
+
+    constructor(
+        IStreamingFeeModule _streamingFeeModule
+    )
+        public
+    {
+        streamingFeeModule = _streamingFeeModule;
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * ANYONE CALLABLE: Accrues fees from streaming fee module. Gets resulting balance after fee accrual, calculates fees for
+     * owner and methodologist, and sends to owner fee recipient and methodologist respectively.
+     */
+    function accrueFeesAndDistribute(ISetToken _setToken) public {
+        // Emits a FeeActualized event
+        streamingFeeModule.accrueFee(_setToken);
+
+        uint256 totalFees = _setToken.balanceOf(address(_manager(_setToken)));
+
+        address methodologist = _manager(_setToken).methodologist();
+        address ownerFeeRecipient = _manager(_setToken).ownerFeeRecipient();
+
+        uint256 ownerTake = totalFees.preciseMul(_manager(_setToken).ownerFeeSplit());
+        uint256 methodologistTake = totalFees.sub(ownerTake);
+
+        if (ownerTake > 0) {
+            _manager(_setToken).transferTokens(address(_setToken), ownerFeeRecipient, ownerTake);
+        }
+
+        if (methodologistTake > 0) {
+            _manager(_setToken).transferTokens(address(_setToken), methodologist, methodologistTake);
+        }
+
+        emit FeesDistributed(address(_setToken), ownerFeeRecipient, methodologist, ownerTake, methodologistTake);
+    }
+
+    /**
+     * ONLY OWNER: Initializes StreamingFeeSplitExtension to the DelegatedManager.
+     *
+     * @param _delegatedManager     Instance of the DelegatedManager to initialize
+     */
+    function initializeExtension(IDelegatedManager _delegatedManager) external {
+        require(msg.sender == _delegatedManager.owner(), "Must be owner");
```

Comment:
So we have really four flows we need to keep in mind here:
1) Full Set deployment and initialization - here we could use the factory to validate that the passed delegated manager belongs to a given SetToken
2) Set migration - again here we can check the factory to see if the Manager/Set is recognized by the factory and being initialized
3) Adding an extension later, module initialized (edge case, very unlikely) - we won't have a factory to check against UNLESS we want to track `validSets` again. Then for initialization all we have to do is check that the factory did deploy the manager.
4) Adding an extension later, module NOT initialized - Same here if we can check a `validSets` state

Reply:
We'll also need some way to check that the manager is legit